### ISSUE TITLE
Schema-First: Can't use enum list as input

### ISSUE THREAD

**Issue by husseinraoouf** [ChilliCream/hotchocolate#2409]
**Describe the bug**
i can't use enum list as input 

#### example:
schema.gql
```gql
input ActionSearchFilter {
  keyword: String
  statuses: [ActionStatus!]
}

enum ActionStatus {
  COMPLETED
  ACTIVE
  OVERDUE
}
```
ActionSearchFilter.cs
```c#
public class ActionSearchFilter
{
    public HotChocolate.Optional<string> keyword { get; set; }
    public HotChocolate.Optional<List<ActionStatus>> statuses { get; set; }
}

public enum ActionStatus
{
    Completed = 1,
    Active = 2,
    Overdue = 3,
}
```
startup.cs
```c#
# removed for simplicty
    services
        .AddGraphQLServer()
            # removed for simplicty
            .BindComplexType<ActionSearchFilter>();
# removed for simplicty
```
when i use `ActionSearchFilter` as input for a query i see this error
```json
{
  "error": {
    "errors": [
      {
        "message": "Unexpected Execution Error",
        "locations": [
          {
            "line": 2,
            "column": 3
          }
        ],
        "path": [
          "actionSearch"
        ],
        "extensions": {
          "message": "Unable to convert type from `List`1` to `List`1`",
          "stackTrace": "   at HotChocolate.Utilities.DefaultTypeConverter.Convert(Type from, Type to, Object source)\n   at HotChocolate.Utilities.Serialization.InputObjectParserHelper.ConvertValue(InputField field, ITypeConverter converter, Object value)\n   at HotChocolate.Utilities.Serialization.InputObjectParserHelper.Parse(InputObjectType type, ObjectValueNode source, IDictionary`2 target, ITypeConverter converter)\n   at HotChocolate.Utilities.Serialization.InputObjectParserHelper.ParseLiteral(InputObjectType type, ObjectValueNode value, InputObjectFactory factory, ITypeConverter converter)\n   at HotChocolate.Types.InputObjectType.<>c__DisplayClass31_1.<OnCompleteType>b__4(ObjectValueNode ov)\n   at HotChocolate.Types.InputObjectType.ParseLiteral(IValueNode valueSyntax, Boolean withDefaults)\n   at HotChocolate.Execution.Processing.MiddlewareContext.CoerceArgumentValue[T](ArgumentValue argument)\n   at HotChocolate.Execution.Processing.MiddlewareContext.ArgumentValue[T](NameString name)\n   at HotChocolate.Execution.Processing.MiddlewareContext.Argument[T](NameString name)\n   at lambda_method143(Closure , IResolverContext )\n   at HotChocolate.Types.FieldMiddlewareCompiler.<>c__DisplayClass3_0.<<CreateResolverMiddleware>b__0>d.MoveNext()\n--- End of stack trace from previous location ---\n   at HotChocolate.Execution.Processing.ResolverTask.ExecuteResolverPipelineAsync()\n   at HotChocolate.Execution.Processing.ResolverTask.TryExecuteAsync()"
        }
      }
    ]
  }
}
```

and if i changed `statuses: [ActionStatus!]` to `statuses: ActionStatus` it works as expected

**To Reproduce**
using schema first approach
1. create input type that has a field of type enum list
2. register that type with `BindComplexType` in startup.cs
3. use that query with a valid value to the field with enum list type 

**Expected behavior**
using enum list as a type for a field in input type should work

**Desktop (please complete the following information):**
 - OS: elementary os 5.1
 - Version: 11.0.0-preview.146

**Comment by michaelstaib** [comment]
Thanks for reporting this one.

**Comment by PascalSenn** [comment]
Test case that produces this error:
```csharp

        [Fact]
        public async Task EnumType_Should_WorkAsListInput()
        {
            // arrange
            var schema = Schema.Create(
                @"
                type Query {
                    test(arg: TestInput!): [FooBar!]
                }

                input TestInput {
                  fooBar: [FooBar!]
                }

                enum FooBar {
                  FOO
                  BAR
                }
                ",
                c =>
                {
                    c.BindType<TestInput>();
                    c.BindResolver(x => x.ArgumentValue<TestInput>("arg")).To("Query", "test");
                });

            // act
            IExecutionResult result =
                await schema.MakeExecutable().ExecuteAsync("{ test(arg: {fooBar: [FOO]}) }");

            // assert
            Assert.Null(result.Errors);
            result.MatchSnapshot();
        }

        public class TestInput
        {
            public Optional<List<ActionStatus>> FooBar { get; set; }
        }

        public enum ActionStatus
        {
            Foo = 1,
            Bar = 2,
        }

```

**Comment by michaelstaib** [comment]
This one is now implemented and will be merged into 11.0.0-rc.2.

Here is how it works:

```csharp
// arrange
Snapshot.FullName();

var sdl =
    @"type Query {
        hello(greetings: Greetings): Greetings
    }
    
    enum Greetings {
        GOOD_MORNING_FOO
    }
    
    extend enum Greetings {
        GOOD_EVENING
    }";

// act
// assert
await new ServiceCollection()
    .AddGraphQL()
    .AddDocumentFromString(sdl)
    .BindComplexType<Query>()
    .BindEnumType<Greetings>(c => 
    {
        // Explicit value to name binding
        c.Value(Greetings.GoodMorning).To("GOOD_MORNING_FOO");

        // Explicit binding
        c.Value(Greetings.GoodEvening);

       // Note that all other values would be bound implicitly.
    })
    .ExecuteRequestAsync("{ hello(greetings: GOOD_EVENING) }")
    .MatchSnapshotAsync();
```
